### ISSUE TITLE
Cueing highlight/arrow can be on nonexistent chocolate bar

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/mean-share-and-balance#319]
**Test device**
MacBook Air M1 chip and Dell

**Operating System**
macOS14.5 and Win10

**Browser**
Safari and Chrome

**Problem description**
For https://github.com/phetsims/qa/issues/1105, in the State Wrapper on the Distribute Screen it is possible to have the cueing arrow/highlight present on a nonexistent chocolate bar in the downstream sim. 

**Steps to reproduce**
1. In the State Wrapper set state rate = 0
2. Go to the Distribute Screen
3. Increase the number of bars for person 2 (e.g. to 7)
4. In the notepad area, move some of those bars to person one's plate so their stack is higher and the cueing arrow is on that stack now.
5. Press Set State Now
6. Press Sync button in downstream sim

**Visuals**

https://github.com/phetsims/mean-share-and-balance/assets/87318828/4fb65215-3ed1-449f-b4a6-20a7210c3cb0

**Comment by marlitas** [comment]
Great catch! The `stackChangedEmitter` was not emitting in state setting because it was being skipped in the `tableSnackNumberProperty` link. I switched the emitter to fire from when snacks are added or removed from each plate's observable array. This will fire more often, but also is more accurate with our code philosophy in this sim where the observable arrays are the ground truth and engine for the logic of the sim.

This is ready for @Nancy-Salpepi's review. Feel free to close if all looks well!

**Comment by Nancy-Salpepi** [comment]
Looks great on main! Closing.

**Comment by jbphet** [comment]
Reopening - the firing of `stackChangedEmitter` on the removal of candy bars from plates caused some regressions that went initially unnoticed, but are pretty significant.  @marlitas and I have been working through them today.

**Comment by jbphet** [comment]
@marlitas and I worked on this together for a few hours, and we think we've got it worked out.  I've committed the changes, and am assigning to @marlitas for a bit more testing and review.

**Comment by marlitas** [comment]
When looking at this together @jbphet and I talked about being consistent with cue handling in our sim specific code, however the keyboard sort cue visibility is managed through a DerivedProperty in the GroupSortInteractionModel. I don't think it would be wise to mess with that. This just led to one minor documentation change committed above.

 I did some additional state wrapper testing and I believe this issue is now ready for RC test.
